helper/iofs/iofs.go

```diff
@@ -0,0 +1,142 @@
+// Package iofs provides an adapter from billy.Filesystem to a the
+// standard library io.fs.FS interface.
+package iofs
+
+import (
+	"io"
+	"io/fs"
+	"path/filepath"
+
+	billyfs "github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/helper/polyfill"
+)
+
+// Wrap adapts a billy.Filesystem to a io.fs.FS.
+func Wrap(fs billyfs.Basic) fs.FS {
+	return &adapterFs{fs: polyfill.New(fs)}
+}
+
+type adapterFs struct {
+	fs billyfs.Filesystem
+}
+
+var _ fs.FS = (*adapterFs)(nil)
+var _ fs.ReadDirFS = (*adapterFs)(nil)
+var _ fs.StatFS = (*adapterFs)(nil)
+var _ fs.ReadFileFS = (*adapterFs)(nil)
```

Comment:
Do you mind elaborating on the reasoning behind this please?

Reply:
Sure, these are simply asserting that `adapterFS` implements these various interfaces.  When using gopls, I was able to use this to automatically generate the method skeletons, but it also seems useful to have the compile-time type assertions that the adapter fulfils these interfaces.

Reply:
Thanks for clarifying and adding the comment above.